src/category_theory/limits/shapes/equalizers.lean

```diff
@@ -551,64 +504,40 @@ end
 
 section
 
-/-- The identity determines a cocone on the coequalizer diagram of f and f -/
-def cocone_parallel_pair_self : cocone (parallel_pair f f) :=
-{ X := Y,
-  ι :=
-  { app := λ j, match j with | zero := f | one := 𝟙 Y end,
-    naturality' := λ j j' g,
-    begin
-      cases g,
-      { refl },
-      { erw category.comp_id f },
-      { dsimp, simp }
-    end } }
+/-- The identity determines a cocone on the coequalizer diagram of `f` and `g`, if `f = g`. -/
+def id_cofork (h : f = g) : cofork f g :=
+cofork.of_π (𝟙 Y) $ h ▸ rfl
 
-@[simp] lemma cocone_parallel_pair_self_ι_app_one : (cocone_parallel_pair_self f).ι.app one = 𝟙 Y :=
-rfl
+/-- The identity on `Y` is a coequalizer of `(f, g)`, where `f = g`.  -/
+def is_colimit_id_cofork (h : f = g) : is_colimit (id_cofork f g h) :=
+cofork.is_colimit.mk _
+  (λ s, cofork.π s)
+  (λ s, category.id_comp _)
+  (λ s m h, by { convert h one, exact (category.id_comp _).symm })
```

Comment:
You're just building a structure, so use `{ lift := ... }`. (I think it's slightly easier to read if the field names are visible.)

Reply:
I'm not quite sure what you mean. I am using [this helper function](https://github.com/leanprover-community/mathlib/blob/master/src/category_theory/limits/shapes/equalizers.lean#L276). Building the structure myself would mean extra work.

Reply:
Sorry, my mistake.

(Not necessary to make any change here, but I wonder if we should give helper functions like this more explicit names, e.g. `mk_helper`, to save this sort of confusion.)

Reply:
> (Not necessary to make any change here, but I wonder if we should give helper functions like this more explicit names, e.g. `mk_helper`, to save this sort of confusion.)

I'll keep that in mind for a future PR.